### ISSUE TITLE
Trouble login with IE11

### ISSUE THREAD

**Issue by adrianabreu** [AzureAD/microsoft-authentication-library-for-js#265]
Hi, I'm currently using msal 0.1.5 and I'm unable to log in using internet explorer.

After an exhaustive debugging session I've realised that the custom_scope is not being saved so the current scope is not maching and I'm receiving an error.

In the following code from msal, while in firefox or chrome msal is entering in pendingCallback that never happens on IE. 

```
        var isCallback = this.isCallback(urlHash);
        if (isCallback) {
            this.handleAuthenticationResponse.call(this, urlHash);
        }
        else {
            var pendingCallback = this._cacheStorage.getItem(Constants_1.Constants.urlHash);
            if (pendingCallback) {
                this.processCallBack(pendingCallback);
            }
        }
```

Suggesttions?? @rohitnarula7176

**Comment by jamiefraser** [comment]
I'm seeing the same thing with 0.1.3.  Any hope of getting this addressed?  It's a showstopper for my client

**Comment by rohitnarula7176** [comment]
@adrianabreu If you are facing this issue only in IE, can you try adding all the sites url, including your app's root page, https://login.microsoftonline.com , https://sts.microsoft.com to the trusted sites list in IE's security settings and see if the issue is resolved. Please refer to the link below on instructions on how to do this:
https://github.com/AzureAD/azure-activedirectory-library-for-js/wiki/Known-issues-on-Edge

**Comment by adrianabreu** [comment]
Thank you for answering.

I've been also checking Edge and it fails sometimes. 

I've been able to log in in IE twice after doing that.

It is not a reliable solution and I can't ask all the clients to make that on their browsers.

There is a current branch working on how to deal with this? 


I've been looking to the oidc library hello.js, but I've been unable to make the refresh token work properly without refreshing the window.

**Comment by rohitnarula7176** [comment]
@adrianabreu We do not have a solution other than the one I have mentioned above. This is a issue we are tracking with the browser team. Closing this as this is not an issue with the library.

**Comment by adrianabreu** [comment]
Thanks again for your time.

I understand that in a future the bug will be solved in Edge.

But, as there is no tracking on IE 11, does that mean that we won't be able to get the refresh token working properly on IE 11 without forcing all the customers to add the trusted sites?

**Comment by adrianabreu** [comment]
Hi, could you check my last question @rohitnarula7176 ? Thank you

**Comment by nehaagrawal** [comment]
@adrianabreu @jamiefraser   This is fixed and released in MSAL 0.2.2. Please check release notes for more details.
